demo/site/src/middleware.ts

```diff
@@ -6,32 +6,25 @@ import { withDomainRewriteMiddleware } from "./middleware/domainRewrite";
 import { withPredefinedPagesMiddleware } from "./middleware/predefinedPages";
 import { withPreviewMiddleware } from "./middleware/preview";
 import { withRedirectToMainHostMiddleware } from "./middleware/redirectToMainHost";
+import { withRobotsMiddleware } from "./middleware/robots";
+import { withSkipRewriteMiddleware } from "./middleware/skipRewrite";
 import { withStatusMiddleware } from "./middleware/status";
 
 export default chain([
     withStatusMiddleware,
     withAdminRedirectMiddleware,
+    withSkipRewriteMiddleware,
     withDamRewriteMiddleware,
     withCspHeadersMiddleware,
     withPreviewMiddleware,
     withRedirectToMainHostMiddleware,
+    withRobotsMiddleware,
```

Comment:
I'd add a comment here stating that this has to be done after the main host redirect:

```suggestion
    withRobotsMiddleware, // for robots.txt, the middleware must be skipped after the main host redirect
```

Because the code looks similar to `withSkipRewriteMiddleware` and I already see myself asking why this isn't done in `withSkipRewriteMiddleware` 6 months from now